tests/objects/generics/test_names.py

```diff
@@ -0,0 +1,52 @@
+from typing import Generic, NewType, TypeVar
+
+import pytest
+
+import strawberry
+from strawberry.enum import EnumDefinition
+from strawberry.lazy_type import LazyType
+from strawberry.schema.config import StrawberryConfig
+from strawberry.type import StrawberryList
+from strawberry.union import StrawberryUnion
+
+
+T = TypeVar("T")
+
+
+Enum = EnumDefinition(None, name="Enum", values=[], description=None)  # type: ignore
+CustomInt = strawberry.scalar(NewType("CustomInt", int))
+
+
+@strawberry.type
+class TypeA:
+    name: str
+
+
+@strawberry.type
+class TypeB:
+    age: int
+
+
+@pytest.mark.parametrize(
+    "types,expected_name",
+    [
+        ([StrawberryList(str)], "ListStrExample"),
+        ([StrawberryList(StrawberryList(str))], "ListListStrExample"),
+        ([StrawberryList(Enum)], "ListEnumExample"),
+        ([StrawberryUnion("Union", (TypeA, TypeB))], "UnionExample"),  # type: ignore
+        ([TypeA], "TypeAExample"),
+        ([CustomInt], "CustomIntExample"),
+        ([TypeA, TypeB], "TypeATypeBExample"),
+        ([TypeA, LazyType["TypeB", "test_names"]], "TypeATypeBExample"),  # type: ignore
```

Comment:
@BryceBeagle I've changed the naming to be the old one, so now we generate names like so:

`Paginated[Person]` -> `PersonPaginated`

Which makes sense as we are putting related types close to each other (at least alphabetically). Person always lives near PersonPaginated.

When using optional and string we get this at the moment:

`Paginated[Optional[Person]]` -> `OptionalPersonPaginated`
`Paginated[List[Person]]` -> `ListPersonPaginated`

shall we change it to be:

`Paginated[Optional[Person]]` -> `PersonOptionalPaginated`
`Paginated[List[Person]]` -> `PersonListPaginated`

?

Reply:
updated :)

Reply:
Good thing you decided against mixing prefix and postfix name generation. Otherwise `Optional[List[str]]` and `List[Optional[str]]` would result in the same type name being generated.

Reply:
uh, I should have added a test for this!